base/builtin/info.go

```diff
@@ -81,13 +83,24 @@ func getNicInfo(cmd *pm.Command) (interface{}, error) {
 		ret[i].HardwareAddr = ifc.HardwareAddr
 		ret[i].Flags = ifc.Flags
 		ret[i].Addrs = ifc.Addrs
-		dat, err := ioutil.ReadFile("/sys/class/net/" + ifc.Name + "/speed")
-		if err != nil {
-			speed = 0
-		} else {
-			speed, _ = strconv.ParseInt(strings.Trim(string(dat), "\n"), 10, 64)
-		}
+		// no need to check for erros here, as the file is necessarily there,
+		// as it's read in just before by net.Interfaces
+		dat, _ := ioutil.ReadFile("/sys/class/net/" + ifc.Name + "/speed")
+		speed, _ = strconv.ParseInt(strings.Trim(string(dat), "\n"), 10, 64)
```

Comment:
so we expect the kernel to always return an parsable int here ?

Reply:
Same kind of remark, would be nice to extract all the parsing into function and test them

Reply:
aye, __always__ a number

Reply:
basically we can be sure that a `/sys/` readable entry has certain fixed types (albeit that they are strings)...
but changes in versions of kernel __might__ occur (like below), but then we'll have to adapt...
Even command outputs are not really set in stone (like for hdparm, that doesn't have json output), so parsing these are a bit 'time/version' dependant...

Reply:
> so parsing these are a bit 'time/version' dependent

Exactly, thus the test. good